src/EA.Weee.RequestHandlers/Admin/AddAatfRequestHandler.cs

```diff
@@ -66,7 +66,28 @@ public async Task<bool> HandleAsync(AddAatf message)
                 panArea = await commonDataAccess.FetchLookup<PanArea>(message.Aatf.PanAreaData.Id);
             }
 
-            var aatf = new Domain.AatfReturn.Aatf(
+            if (message.AatfId == null)
```

Comment:
Rather than doing this here is it possible to just pass the AATF id in the constructor and if its null then set it

Reply:
Done